### PR TITLE
Feature/tasks/420135 using external payment

### PR DESCRIPTION
**Pull request by paulshimmen**
This code uses the external payment Id within the return url we send to Gov Pay. The external payment Id is then used to complete the Gov Pay Journey.

### CODE REVIEW

File: src/EPR.Payment.Facade.Common.UnitTests/Mappings/AutoMapperTests.cs [hunk 1]
```diff
@@ -32,39 +33,43 @@ public void AutoMapper_Configuration_IsValid()
             configAction.Should().NotThrow();
         }
 
-        [TestMethod, AutoMoqData]
-        public void PaymentRequestDto_To_GovPayPaymentRequestDto_Mapping_IsValid(PaymentRequestDto paymentRequestDto)
+        [TestMethod]
+        public void PaymentRequestDto_To_GovPayRequestDto_Mapping_IsValid()
         {
-            var govPayPaymentRequestDto = _mapper.Map<GovPayRequestDto>(paymentRequestDto);
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade.Common.UnitTests/Mappings/AutoMapperTests.cs:38]: use automoqdata attribute;
image

File: src/EPR.Payment.Facade.Common.UnitTests/Mappings/AutoMapperTests.cs [hunk 2]
```diff
@@ -32,39 +33,43 @@ public void AutoMapper_Configuration_IsValid()
...
+            var fixture = new Fixture();
+            var paymentRequestDto = fixture.Create<PaymentRequestDto>();
+            var govPayRequestDto = _mapper.Map<GovPayRequestDto>(paymentRequestDto);
 
             using (new FluentAssertions.Execution.AssertionScope())
             {
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade.Common.UnitTests/Mappings/AutoMapperTests.cs:44]: we can import the namespace;
 using FluentAssertions.Execution;
 
and then initialise AssertionScope with the class name only;
image

File: src/EPR.Payment.Facade.Common.UnitTests/Mappings/AutoMapperTests.cs [hunk 3]
```diff
@@ -32,39 +33,43 @@ public void AutoMapper_Configuration_IsValid()
...
+                govPayRequestDto.return_url.Should().BeNull(); // Ignored in mapping
+                govPayRequestDto.Description.Should().BeNull(); // Ignored in mapping
             }
         }
 
-        [TestMethod, AutoMoqData]
-        public void PaymentRequestDto_To_InsertPaymentRequestDto_Mapping_IsValid(PaymentRequestDto paymentRequestDto)
+        [TestMethod]
+        public void PaymentRequestDto_To_InsertPaymentRequestDto_Mapping_IsValid()
         {
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade.Common.UnitTests/Mappings/AutoMapperTests.cs:70]: Use AutoMoqdata to inject the dependencies wherever possible. See the comment above

File: src/EPR.Payment.Facade.Common/Dtos/Request/Payments/CompletePaymentRequestDto.cs [hunk 4]
```diff
@@ -1,17 +1,14 @@
-﻿using Newtonsoft.Json;
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade.Common/Dtos/Request/Payments/CompletePaymentRequestDto.cs:1]: can we delete this file if the class is not used

File: src/EPR.Payment.Facade.UnitTests/Controllers/Payments/PaymentsControllerTests.cs [hunk 5]
```diff
@@ -19,15 +18,15 @@ namespace EPR.Payment.Facade.UnitTests.Controllers
     [TestClass]
     public class PaymentsControllerTests
     {
-        private IFixture? fixture;
+        private IFixture? _fixture;
 
         [TestInitialize]
         public void Initialize()
         {
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade.UnitTests/Controllers/Payments/PaymentsControllerTests.cs:25]: The initialize method is not needed. There is no point creating a fixture as it is not used.

File: src/EPR.Payment.Facade.UnitTests/Controllers/Payments/PaymentsControllerTests.cs [hunk 6]
```diff
@@ -83,10 +82,14 @@ public void Initialize()
                 It.Is<Func<It.IsAnyType, Exception?, string>>((v, t) => true)),
             Times.Once);
 
-            var contentResult = result as ContentResult;
-            contentResult?.StatusCode.Should().Be(StatusCodes.Status200OK);
-            contentResult?.ContentType.Should().Be("text/html");
-            contentResult?.Content.Should().Contain("window.location.href = 'https://example.com/error'");
+            using (new FluentAssertions.Execution.AssertionScope())
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade.UnitTests/Controllers/Payments/PaymentsControllerTests.cs:85]: See my earlier comment on not using the full qualified name in assertionscope. Please do it in all the tets

File: src/EPR.Payment.Facade/Controllers/Payments/PaymentsController.cs [hunk 7]
```diff
@@ -100,32 +100,32 @@ public async Task<IActionResult> InitiatePayment([FromBody] PaymentRequestDto re
         }
     }
 
-    [HttpPost("{govPayPaymentId}/complete")]
+    [HttpPost("{externalPaymentId}/complete")]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade/Controllers/Payments/PaymentsController.cs:106]: The method is marked to produce 500InternalServerError but the code does not return this response code in any condition. Remove this if not required.

**Review by manishagarwalEviden — CHANGES_REQUESTED** [review]

File: src/EPR.Payment.Facade.Common.UnitTests/Mappings/AutoMapperTests.cs (hunk 1 above)
Comment (paulshimmen) [reply src/EPR.Payment.Facade.Common.UnitTests/Mappings/AutoMapperTests.cs:38]: Sure will do.

File: src/EPR.Payment.Facade.Common.UnitTests/Mappings/AutoMapperTests.cs (hunk 2 above)
Comment (paulshimmen) [reply src/EPR.Payment.Facade.Common.UnitTests/Mappings/AutoMapperTests.cs:44]: I have changed all tests and now import the namespace

File: src/EPR.Payment.Facade.Common.UnitTests/Mappings/AutoMapperTests.cs (hunk 3 above)
Comment (paulshimmen) [reply src/EPR.Payment.Facade.Common.UnitTests/Mappings/AutoMapperTests.cs:70]: Sure will do.

File: src/EPR.Payment.Facade.Common/Dtos/Request/Payments/CompletePaymentRequestDto.cs (hunk 4 above)
Comment (paulshimmen) [reply src/EPR.Payment.Facade.Common/Dtos/Request/Payments/CompletePaymentRequestDto.cs:1]: Sure, sorry, meant to do this. I will delete.

File: src/EPR.Payment.Facade.UnitTests/Controllers/Payments/PaymentsControllerTests.cs (hunk 5 above)
Comment (paulshimmen) [reply src/EPR.Payment.Facade.UnitTests/Controllers/Payments/PaymentsControllerTests.cs:25]: Agreed, I will remove and make sure automoqdata is being used.

File: src/EPR.Payment.Facade.UnitTests/Controllers/Payments/PaymentsControllerTests.cs (hunk 6 above)
Comment (paulshimmen) [reply src/EPR.Payment.Facade.UnitTests/Controllers/Payments/PaymentsControllerTests.cs:85]: Sure, I will change for all tets.

File: src/EPR.Payment.Facade/Controllers/Payments/PaymentsController.cs (hunk 7 above)
Comment (paulshimmen) [reply src/EPR.Payment.Facade/Controllers/Payments/PaymentsController.cs]: I have updated the ProducesResponseTypes for InitiatePayment and CompletePayment so they accurately reflect the possible response types.

**Review by manishagarwalEviden — APPROVED** [review]

**Pull request merged**